opensearchpy/connection/http_requests.py

```diff
@@ -187,9 +190,14 @@ def perform_request(  # type: ignore
             "allow_redirects": allow_redirects,
         }
         send_kwargs.update(settings)
+        self.is_metrics_none = False
```

Comment:
This looks like an anti-pattern. Either `self.metrics` is set, or it's not. Default to not.

Reply:
@dblock, If it's not used, then [self.metrics.request_start()](https://github.com/opensearch-project/opensearch-py/pull/689/files#diff-1e38589c63abdaf216eafb88d5419b81729acb2c5ccc71d5b4af7a52da105e47R277) and [self.metrics.request_end()](https://github.com/opensearch-project/opensearch-py/pull/689/files#diff-1e38589c63abdaf216eafb88d5419b81729acb2c5ccc71d5b4af7a52da105e47R281) can't be calculated. Are you suggesting we keep [start = time.time()](https://github.com/opensearch-project/opensearch-py/pull/689/files#diff-1e38589c63abdaf216eafb88d5419b81729acb2c5ccc71d5b4af7a52da105e47L245) and [duration = time.time() - start](https://github.com/opensearch-project/opensearch-py/pull/689/files#diff-1e38589c63abdaf216eafb88d5419b81729acb2c5ccc71d5b4af7a52da105e47L274) untouched, and only calculate and return service_time when self.metrics is not None?

Reply:
Made the changes as requested. Working on adding tests.